modules/sor/sorV2/lib/poolsV3/gyroECLP/gyroECLPPool.ts

```diff
@@ -0,0 +1,394 @@
+            const token = new Token(
+                parseFloat(chainToIdMap[pool.chain]),
+                poolToken.address as Address,
+                poolToken.token.decimals,
+                poolToken.token.symbol,
+                poolToken.token.name,
+            );
+            const scale18 = parseEther(poolToken.balance);
+            const tokenAmount = TokenAmount.fromScale18Amount(token, scale18);
+
+            if (poolToken.token.underlyingTokenAddress) {
+                const underlyingToken = underlyingTokens.find(
+                    (token) => token.address === poolToken.token.underlyingTokenAddress,
+                );
+                if (underlyingToken) {
+                    const unwrapRateDecimals = 18 - poolToken.token.decimals + underlyingToken.decimals;
+                    poolTokens.push(
+                        new Erc4626PoolToken(
+                            token,
+                            tokenAmount.amount,
+                            poolToken.index,
+                            parseEther(poolToken.priceRate),
+                            parseUnits(poolToken.token.unwrapRate, unwrapRateDecimals),
+                            poolToken.token.underlyingTokenAddress,
+                        ),
+                    );
+                } else {
+                    poolTokens.push(
+                        new PoolTokenWithRate(
+                            token,
+                            tokenAmount.amount,
+                            poolToken.index,
+                            parseEther(poolToken.priceRate),
+                        ),
+                    );
+                }
+            } else {
+                poolTokens.push(
+                    new PoolTokenWithRate(token, tokenAmount.amount, poolToken.index, parseEther(poolToken.priceRate)),
+                );
+            }
+        }
+
+        const totalShares = parseEther(pool.dynamicData.totalShares);
+
+        const gyroData = pool.typeData as GyroData;
+
+        const gyroEParams: GyroEParams = {
+            alpha: parseEther(gyroData.alpha),
+            beta: parseEther(gyroData.beta),
+            c: parseEther(gyroData.c!),
+            s: parseEther(gyroData.s!),
+            lambda: parseEther(gyroData.lambda!),
+        };
+
+        const derivedGyroEParams: DerivedGyroEParams = {
+            tauAlpha: {
+                x: parseUnits(gyroData.tauAlphaX!, 38),
+                y: parseUnits(gyroData.tauAlphaY!, 38),
+            },
+            tauBeta: {
+                x: parseUnits(gyroData.tauBetaX!, 38),
+                y: parseUnits(gyroData.tauBetaY!, 38),
+            },
+            u: parseUnits(gyroData.u!, 38),
+            v: parseUnits(gyroData.v!, 38),
+            w: parseUnits(gyroData.w!, 38),
+            z: parseUnits(gyroData.z!, 38),
+            dSq: parseUnits(gyroData.dSq!, 38),
```

Comment:
Curious about the need for non null assertions here? 

Removing doesn't seem to make the type checker angry

Reply:
Also the derived params are stored on chain as 38 decimal values, so why the `parseUnits`?

Reply:
Api seems to be storing values as floating point numbers
```
query {
  poolGetPool(
    id: "0x80fd5bc9d4fa6c22132f8bb2d9d30b01c3336fb3"
    chain:SEPOLIA
  ) {
  	id
    type
    ... on GqlPoolGyro {
     beta
     tauAlphaX
    }
  }
}
```

Reply:
yeah, that's why we need to parse them back to fixed point numbers

Reply:
And non null assertions were copied from the v2 implementation - they were necessary on the old api implementation, but that's no longer the case.
Thanks for pointing it out - I'll remove them here.